Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -450,97 +464,123 @@ class HistoryPanelSiteTableViewController: SiteTableViewController {
         // Intentionally blank. Required to use UITableViewRowActions
     }
 
-    func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [AnyObject]? {
-        let title = NSLocalizedString("Remove", tableName: "HistoryPanel", comment: "Action button for deleting history entries in the history panel.")
-
-        let delete = UITableViewRowAction(style: UITableViewRowActionStyle.Default, title: title, handler: { (action, indexPath) in
-            if let site = self.siteForIndexPath(indexPath) {
-                // Why the dispatches? Because we call success and failure on the DB
-                // queue, and so calling anything else that calls through to the DB will
-                // deadlock. This problem will go away when the history API switches to
-                // Deferred instead of using callbacks.
-                self.profile.history.removeHistoryForURL(site.url)
-                    .upon { res in
-                        self.fetchData().uponQueue(dispatch_get_main_queue()) { result in
-                            // If a section will be empty after removal, we must remove the section itself.
-                            if let data = result.successValue {
-
-                                let oldCategories = self.categories
-                                self.data = data
-                                self.computeSectionOffsets()
-
-                                let sectionsToDelete = NSMutableIndexSet()
-                                var rowsToDelete = [NSIndexPath]()
-                                let sectionsToAdd = NSMutableIndexSet()
-                                var rowsToAdd = [NSIndexPath]()
-
-                                for (index, category) in self.categories.enumerate() {
-                                    let oldCategory = oldCategories[index]
-
-                                    // don't bother if we're not displaying this category
-                                    if oldCategory.section == nil && category.section == nil {
-                                        continue
-                                    }
+    private func removeHistoryForURL(indexPath: NSIndexPath) {
```

Comment:
Yeah look at it now it makes more sense to name it `removeHistoryForURLAtIndexPath`